lib/src/rating/rating.component.ts

```diff
@@ -0,0 +1,231 @@
+import { Component, Input, forwardRef, ViewEncapsulation, ViewChildren, QueryList, ElementRef, OnInit } from "@angular/core";
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+
+export const CUSTOM_RATING_VALUE_ACCESSOR: any = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => RatingComponent),
+    multi: true,
+};
+
+@Component({
+    selector: "ac-rating",
+    templateUrl: "./rating.component.html",
+    styleUrls: ["./rating.component.scss"],
+    providers: [CUSTOM_RATING_VALUE_ACCESSOR],
+    encapsulation: ViewEncapsulation.None,
+})
+export class RatingComponent implements ControlValueAccessor, OnInit {
+    @Input() iconWidth?: number = 25;
+    @Input() iconHeight?: number = 25;
+    @Input() max?: number = 5;
+    @Input() readOnly?: boolean = false;
+    @Input() tooltipList?: Array<string>;
+    @Input() className?: string;
+    @Input() useHollow?: boolean;
+    @Input() showValue?: boolean;
+    @Input() showTextValue?: boolean;
+
+    randomIds: Array<string> = [];
+
+    /**
+     * @member 0 Grey (unselected)
+     * @member 1 Yellow (Selected)
+     */
+    @Input() colors?: Array<string> = ["#A9A9A9", "#ffc500"];
+
+    // Placeholders for the callbacks which are later provided
+    // by the Control Value Accessor
+    private innerValue: any = "";
+
+    activeList: Array<string> = [];
+    initialized: boolean = false;
+    displayValue?: string;
+
+    tabCounter: number = 0;
+
+    @ViewChildren("radioItemRefs") radios: QueryList<ElementRef>;
+
+    ngOnInit() {
+        if (this.max) {
+            this.randomIds = Array.apply(null, Array(this.max)).map((item: any) => {
+                const identifier = +Math.floor(Math.random() * 100) + new Date().getTime();
+                return "star-" + identifier;
+            });
+        }
+    }
+
+    /**
+     * Provides an array for ngFor to loop through
+     * @returns {Array<number>} An array of ones just to loop through it
+     */
+    getList(): Array<number> {
+        const list: Array<number> = new Array(this.max);
+        for (let i: number = 0; i < list.length; i++) {
+            list[i] = 1;
+        }
+        return list;
```

Comment:
```suggestion
        return Array(this.max).fill(1);
```

Reply:
Not necessary.

Reply:
It's OK, we have polyfills for that, I just don't think the function is necessary. If you don't like it you can even use this:
```javascript
"1".repeat(this.max).split("").map(i => +i)
```